ludwig/features/image_feature.py

```diff
@@ -166,110 +178,123 @@ def add_feature_data(
-            raise ValueError(
-                'Image file paths must be absolute'
-            )
-
-        first_image = imread(
-            get_abs_path(
-                csv_path,
-                dataset_df[feature['name']][0]
-            )
-        )
-
-        first_img_height = first_image.shape[0]
-        first_img_width = first_image.shape[1]
-        first_img_num_channels = num_channels_in_image(first_image)
-
-        if height == 0 or width == 0:
-            # User hasn't specified height and width
+        else:
+            # User hasn't specified height and width.
+            # So we assume that all images have the same width and height.
+            # Thus the width and height of the first one are the same
+            # as all the other ones
             height = first_img_height
             width = first_img_width
 
-        # User specified num_channels in the model/feature definition
-        user_specified_num_channels = False
-        num_channels = first_img_num_channels
         if NUM_CHANNELS in preprocessing_parameters:
+            # User specified num_channels in the model/feature definition
             user_specified_num_channels = True
             num_channels = preprocessing_parameters[NUM_CHANNELS]
+        else:
+            user_specified_num_channels = False
+            num_channels = first_img_num_channels
 
         assert isinstance(num_channels, int), ValueError(
-            'Number of image channels needs to be an integer')
+            'Number of image channels needs to be an integer'
+        )
+
+        return (should_resize,
+                width,
+                height,
+                num_channels,
+                user_specified_num_channels,
+                first_image)
+
+    @staticmethod
+    def add_feature_data(
+            feature,
+            dataset_df,
+            data,
+            metadata,
+            preprocessing_parameters
+    ):
+        set_default_value(
+            feature['preprocessing'],
+            'in_memory',
+            preprocessing_parameters['in_memory']
+        )
+        set_default_value(
+            feature['preprocessing'],
+            'num_processes',
+            preprocessing_parameters['num_processes']
+        )
+        csv_path = None
+        if hasattr(dataset_df, 'csv'):
+            csv_path = os.path.dirname(os.path.abspath(dataset_df.csv))
+
+        num_images = len(dataset_df)
+        if num_images == 0:
+            raise ValueError('There are no images in the dataset provided.')
+
+        first_image_path = dataset_df[feature['name']][0]
+        if csv_path is None and not os.path.isabs(first_image_path):
+            raise ValueError('Image file paths must be absolute')
+
+        first_image_path = get_abs_path(csv_path, first_image_path)
+
+        (should_resize,
+         width,
+         height,
+         num_channels,
+         user_specified_num_channels,
+         first_image) = \
```

Comment:
Nit: avoid using `\` for line breaks - these are not PEP8 compliant. Use parentheses instead, e.g.
y = (
    long_funciton_name()
)